examples_utils/benchmarks/run_benchmarks.py

```diff
@@ -152,6 +153,47 @@ def proc_thread():
     return (output, err, exitcode)
 
 
+def install_patched_requirements(requirements_file: Union[str, Path], listener: TextIOWrapper):
+    """Removes any 'examples-utils' requirements from a a requirements
+    file before installing it. It returns the original unpatched requirements
+    in case they are needed later."""
+
+    requirements_file = Path(requirements_file)
+    logger.info(f"Install python requirements")
+    if not requirements_file.exists():
+        raise FileNotFoundError(f"Invalid python requirements where specified at {requirements_file}")
+    # Strip examples-utils requirement as it can break the installation
+    original_requirements = requirements_file.read_text()
+    requirements_file.write_text("\n".join(l for l in original_requirements.splitlines() if "examples-utils" not in l))
+    cmd = [sys.executable, "-m", "pip", "install", "-r", str(requirements_file)]
+    out, err, exit_code = run_and_monitor_progress(cmd, listener)
+    if exit_code:
+        raise subprocess.CalledProcessError(exit_code, cmd, out, err)
+    return original_requirements
+
+
+def install_apt_packages(requirements_file_or_list: Union[str, Path, List[str]], listener: TextIOWrapper):
```

Comment:
Why are we specifying apt requirements in the benchmark? I thought these would go in a required_apt_packages.txt in the example root dir like they do for pytorch cnns? This means the user should install them as part of the make step or intentionally.

I guess its a nice additional bit of automation but so I'm not against it, just curious as to why we are doing this? Do notebook tutorials always need these? Or is this for edge cases? If its the latter, wouldn't it be best to make it a seperate reuirements file? Then again, I can see how it would be a bit of a pain to ask users to install those seperately for specific entries in the benchmarks.yml file.

Reply:
The short answer is "it was the fastest/cleanest way to solve the problem I was encountering" but I'm not sure it makes sense for examples-utils to do this.

If the goal is to make all things examples easier this might make sense as a feature, if it is meant as a narrow tool for running benchmarks, not so much.

Reply:
Yh I dont think it should be in examples utils tbh, sometimes the ordering of installing apt vs pip requirements changes like for PT resnet, its not arbitrary and so anyone who installs pip requirements (to get to examples utils) and then uses this to automatically install apt requirements will break libjpegTurbo (it requires apt before pip I think)

I would avoid adding this into master